supplementary/alica_ros1/alica_ros_turtlesim/test/test_main.cpp

```diff
@@ -0,0 +1,178 @@
+        } while (duration_cast<milliseconds>(system_clock::now() - start).count() < ms);                                                                       \
+        EXPECT_TRUE(turtle.isPoseEqual(past_pose, 0.01));                                                                                                      \
+    } while (0)
+
+// class to handle callbacks
+class TurtlePosition
+{
+public:
+    TurtlePosition() = default;
+
+    TurtlePosition(const double x, const double y, const double theta = 0)
+    {
+        _pose.x = x;
+        _pose.y = y;
+        _pose.theta = theta;
+    }
+
+    void poseCallback(const turtlesim::Pose::ConstPtr msg)
+    {
+        _pose.x = msg->x;
+        _pose.y = msg->y;
+        _pose.theta = msg->theta;
+    }
+
+    turtlesim::Pose getPose() const { return _pose; }
+
+    bool isNear(const turtlesim::Pose& desired_pose, const turtlesim::Pose& delta) const
+    {
+        assert(delta.x > 0);
+        assert(delta.y > 0);
+
+        return std::fabs(_pose.x - desired_pose.x) < delta.x && std::fabs(_pose.y - desired_pose.y) < delta.y;
+    }
+
+    double distance(const TurtlePosition& turtle) { return sqrt(pow(turtle._pose.x - _pose.x, 2) + pow(turtle._pose.y - _pose.y, 2)); }
+
+    bool isPoseEqual(const turtlesim::Pose& other_pose, double tolerance)
+    {
+        return std::fabs(other_pose.x - _pose.x) + std::fabs(other_pose.y - _pose.y) + std::fabs(other_pose.theta - _pose.theta) < tolerance;
+    }
+
+private:
+    turtlesim::Pose _pose;
+};
+
+TEST(AlicaTurtlesimTest, destinationTest)
+{
+    ros::NodeHandle nh;
+    ros::Publisher join_formation_turtle_1_pub = nh.advertise<std_msgs::Empty>("/turtle1/join_formation", 10);
+    ros::Publisher join_formation_turtle_2_pub = nh.advertise<std_msgs::Empty>("/turtle2/join_formation", 10);
+    ros::Publisher join_formation_turtle_3_pub = nh.advertise<std_msgs::Empty>("/turtle3/join_formation", 10);
+    ros::Publisher join_formation_turtle_4_pub = nh.advertise<std_msgs::Empty>("/turtle4/join_formation", 10);
+
+    // wait for subscribers to latch
+    RUN_UNTIL_EXPECT_EQ(join_formation_turtle_1_pub.getNumSubscribers(), 1U, 10000);
+    RUN_UNTIL_EXPECT_EQ(join_formation_turtle_2_pub.getNumSubscribers(), 1U, 10000);
+    RUN_UNTIL_EXPECT_EQ(join_formation_turtle_3_pub.getNumSubscribers(), 1U, 10000);
+    RUN_UNTIL_EXPECT_EQ(join_formation_turtle_4_pub.getNumSubscribers(), 1U, 10000);
+
+    turtlesim::Pose pose_tolerance;
+    pose_tolerance.x = 0.2;
+    pose_tolerance.y = 0.2;
+
+    // Turtle 1 test
+    TurtlePosition turtle1;
+    ros::Subscriber turtle1_sub = nh.subscribe("turtle1/pose", 10, &TurtlePosition::poseCallback, &turtle1);
+
+    turtlesim::Pose desired_pose_turtle_1;
+    desired_pose_turtle_1.x = 4.97;
+    desired_pose_turtle_1.y = 5.033;
+
+    // send init message to turtles
+    std_msgs::Empty msg;
+    join_formation_turtle_1_pub.publish(msg);
+
+    std::this_thread::sleep_for(std::chrono::seconds(10));
```

Comment:
Seems unnecessary, we already allow 10 seconds for robot to stop in the next line

Reply:
Oh sorry, I forgot to remove this, I included in a past commit to see if CI would pass because I was thinking it was something else, but this was implemented already in RUN_UNTIL_ROBOT_STOP